### PR TITLE
QEAUTO-122 DBP-Agent - Check the swicth for Regulated/Unregulated wal…

### PR DESCRIPTION
**Pull request by idt-zerg**
…lets

- Complete scenario
- Add new validation for QEAUTO-95

### CODE REVIEW

File: IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs [hunk 1]
```diff
@@ -107,5 +107,17 @@ public bool IsDefaultWalletSelected()
         {
             return this.Driver.FindElement(this.DefaultWalletExist).Selected;
         }
+
+        /// <summary>
+        /// Verify that element exist
+        /// </summary>
+        /// <param name="lblBalance"></param>
+        /// <param name="value"></param>
+        /// <returns>A boolean according to the presence of the element</returns>
```
Comment (virginiaSanabria) [inline IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs]: improve i.e. /// <returns>True if balance label is displayed.</returns>

File: IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs [hunk 2]
```diff
@@ -107,5 +107,17 @@ public bool IsDefaultWalletSelected()
...
+        public bool IsLblBalanceExist(string lblBalance, string value)
```
Comment (virginiaSanabria) [inline IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs]: I think tha is better IsBalanceLabelDisplayed

File: IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs [hunk 3]
```diff
@@ -107,5 +107,17 @@ public bool IsDefaultWalletSelected()
         {
             return this.Driver.FindElement(this.DefaultWalletExist).Selected;
         }
+
+        /// <summary>
+        /// Verify that element exist
+        /// </summary>
+        /// <param name="lblBalance"></param>
```
Comment (virginiaSanabria) [inline IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs]: miss a little description, the same for next line

File: IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs [hunk 4]
```diff
@@ -107,5 +107,17 @@ public bool IsDefaultWalletSelected()
...
+        /// <param name="value"></param>
+        /// <returns>A boolean according to the presence of the element</returns>
+        public bool IsLblBalanceExist(string lblBalance, string value)
+        {
+            By LblBalanceExist = Utils.GetByLocator("HomeLocators", "LblBalanceExist", lblBalance, value);
```
Comment (virginiaSanabria) [inline IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs]: please review if it is posible to use the IsDiplayed method in core

File: IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+Scenario: C3216 - Check the swicth for Regulated/Unregulated wallets (Agent)
+
+	Given I am login on the portal as a distributor
+	When I switch to "Regulated" Wallet
+	Then I expect can see the switch for "Regulated"
+	And Displays the field "Available Credit" with its balance in "$37,647.07"
+	
+	When I switch to "Default" Wallet
+	Then I expect can see the switch for "Default"
+	And Displays the field "Available Credit" with its balance in "$4,922.19"
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature]: Add a space tab, also you should use 'Verify' or should in the Then steps

File: IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+@smoke
+Scenario: C3216 - Check the swicth for Regulated/Unregulated wallets (Agent)
+
+	Given I am login on the portal as a distributor
+	When I switch to "Regulated" Wallet
+	Then I expect can see the switch for "Regulated"
+	And Displays the field "Available Credit" with its balance in "$37,647.07"
+	
+	When I switch to "Default" Wallet
+	Then I expect can see the switch for "Default"
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature]: you should use 'Verify' or should in the Then steps, i.e. I should see ... or Verify that switch .... is diplayed

File: IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature [hunk 7]
```diff
@@ -1,10 +1,12 @@
 ﻿Feature: Wallets
 @smoke
-Scenario: C3189 - Edit Check the swicth for Regulated/Unregulated wallets
+Scenario: C3189 - Edit Check the swicth for Regulated/Unregulated wallets (Retailer)
 
 	Given I am login on the portal as a distributor
 	When I switch to "Regulated" Wallet
 	Then I expect can see the switch for "Regulated"
+	And Displays the field "Available Credit" with its balance in "$524,915.86"
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature]: Use Verify or should

File: IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature [hunk 8]
```diff
@@ -1,10 +1,12 @@
...
 	
 	When I switch to "Default" Wallet
 	Then I expect can see the switch for "Default"
+	And Displays the field "Available Credit" with its balance in "$2,422.51"
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature]: Use Verify or should

File: IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿Feature: Wallets
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/Features/Agent/SwitchRegulatedUnregulatedWallets.feature]: I know that the features in Xgen project are ordering by  portals , so it and C3189 should be into Retailer folder in the same feature file,please coordinate with Qe_Debt team :David, Ale,....

File: IDT.XGen/Tests/UI/StepDefinitions/Distributor/RegisterRetailersSteps.cs [hunk 10]
```diff
@@ -107,5 +107,11 @@ public void ThenIExpectCanSeeTheSwitchFor(string switchTo)
                     throw new ArgumentOutOfRangeException("are not present (Defaul | Regulated)");
             }
         }
+
+        [Then(@"Displays the field ""(.*)"" with its balance in ""(.*)""")]
+        public void ThenDisplaysTheFieldWithItsBalanceIn(string p0, string p1)
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/StepDefinitions/Distributor/RegisterRetailersSteps.cs]: improve the variable's name , they are not clear

File: IDT.XGen/Tests/UI/StepDefinitions/Distributor/RegisterRetailersSteps.cs [hunk 11]
```diff
@@ -107,5 +107,11 @@ public void ThenIExpectCanSeeTheSwitchFor(string switchTo)
...
+        {
+            Assert.True(this.homePage.IsLblBalanceExist(p0, p1));
```
Comment (virginiaSanabria) [inline IDT.XGen/Tests/UI/StepDefinitions/Distributor/RegisterRetailersSteps.cs]: I should use the Assert.True with message in order to get a clear message when the test will fail, i.e  Assert.True(this.homePage.IsLblBalanceExist(p0, p1), "The balance label doesn't exist in Home page");

File: IDT.XGen/IDT.XGen.csproj [hunk 12]
```diff
@@ -331,10 +335,22 @@
...
   <ItemGroup>
-    <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
-    <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
+    <ProjectReference Include="..\..\IDTCORE1014\IDT.Core.Automation\IDT.Core.Automation.csproj">
+      <Project>{3ac4459c-811a-4878-9f9b-e27ea0dbd1cc}</Project>
+      <Name>IDT.Core.Automation</Name>
+    </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
```
Comment (wMercadoJala) [inline IDT.XGen/IDT.XGen.csproj]: This target should be removed

File: IDT.XGen/IDT.XGen.csproj [hunk 13]
```diff
@@ -331,10 +335,22 @@
     <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
   </ItemGroup>
   <ItemGroup>
-    <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
-    <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
+    <ProjectReference Include="..\..\IDTCORE1014\IDT.Core.Automation\IDT.Core.Automation.csproj">
```
Comment (wMercadoJala) [inline IDT.XGen/IDT.XGen.csproj]: Why CoreAutomation was added to xGen

File: IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs [hunk 14]
```diff
@@ -102,10 +102,22 @@ public bool IsRegulatedWalletSelected()
...
         /// </summary>
-        /// <returns>A boolean according to the presence of the element</returns>
+        /// <returns>True if element is selected, false if not</returns>
         public bool IsDefaultWalletSelected()
         {
             return this.Driver.FindElement(this.DefaultWalletExist).Selected;
         }
+
+        /// <summary>
+        /// Verify that element exist
```
Comment (wMercadoJala) [inline IDT.XGen/Main/UI/Pages/Distributor/HomePage.cs]: Change verb to third person

File: IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature [hunk 15]
```diff
@@ -0,0 +1,24 @@
+﻿Feature: Wallets
+@smoke
+Scenario: C3189 - Edit Check the swicth for Regulated/Unregulated wallets (Retailer)
```
Comment (wMercadoJala) [inline IDT.XGen/Tests/UI/Features/Retailer/SwitchRegulatedUnregulatedWallets.feature]: Is this the same that Fixed Code Review PR ? :S

**Pull request closed without merging**